✨ Publish New Docs ✨ (#4904)
software-mansion/react-native-reanimated: docs/docs/advanced/useFrameCallback.mdx

---
sidebar_position: 3
---

# useFrameCallback

`useFrameCallback` lets you run a function on every frame update.

import FrameCallbackSimple from '@site/src/examples/FrameCallbackSimple';
import FrameCallbackSimpleSrc from '!!raw-loader!@site/src/examples/FrameCallbackSimple';

<InteractiveExample
  src={FrameCallbackSimpleSrc}
  component={<FrameCallbackSimple />}
/>

## Reference

```javascript
import { useFrameCallback } from 'react-native-reanimated';

function App() {
  const frameCallback = useFrameCallback((frameInfo) => {
    // Increment a value on every frame update
    sv.value += 1;
  });

  return (
    <Button
      title="Start/Stop"
      onPress={() => frameCallback.setActive(!frameCallback.isActive)}
    />
  );
}
```

<details>
<summary>Type definitions</summary>

```typescript
type FrameInfo = {
  timestamp: number;
  timeSincePreviousFrame: number | null;
  timeSinceFirstFrame: number;
};

type FrameCallback = {
  setActive: (isActive: boolean) => void;
  isActive: boolean;
  callbackId: number;
};

function useFrameCallback(
  callback: (frameInfo: FrameInfo) => void,
  autostart = true
): FrameCallback;
```

</details>

### Arguments

#### `callback`

> A function to be executed on every frame update. This function receives a `frameInfo` object containing fields:

Reviewer: ```suggestion
A function executed on every frame update. This function receives a `frameInfo` object containing the following fields:
```
